### ISSUE TITLE
[torch.compile][jagged nested tensor creation] AttributeError: 'torch._C._SymNode' object has no attribute 'expr'

### ISSUE THREAD

**Issue by zejun-chen** [pytorch/pytorch#125560]
### 🐛 Describe the bug

Hi,

We can compile the aten ops with the jagged nested tensor successfully, while we got the runtime error when compiling the jagged nested tensor creation op. Below is the reproducer and the associated error we got in torch.compile. It looks like the failure comes from the symbolic shape checking.

```
import torch

def func3(a, b):
    x = torch.nested.nested_tensor([a, b], layout=torch.jagged)
    x = x + 1.0
    x = x * 2.0
    x = x + 3.0
    x = torch.relu(x)
    return torch.abs(x)

a = torch.randn(3,3,device='cuda')
b = torch.randn(3,3,device='cuda')
compiled_func3 = torch.compile(func3)
c = compiled_func3(a, b)
print('nested value = ', nested.values())
print('c      value = ', c.values())

```

Here is the error message:
```
Traceback (most recent call last):
  File "/home/pt-gpu/zejun/nested/nested.py", line 39, in <module>
    c = compiled_func3(a, b)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/eval_frame.py", line 390, in _fn
    return fn(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 945, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state, skip=1)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 808, in _convert_frame
    result = inner_convert(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 400, in _convert_frame_assert
    return _compile(
  File "/home/pt-gpu/anaconda3/envs/zejun/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 692, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/utils.py", line 265, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 545, in compile_inner
    out_code = transform_code_object(code, transform)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/bytecode_transformation.py", line 1036, in transform_code_object
    transformations(instructions, code_options)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 165, in _fn
    return fn(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 507, in transform
    tracer.run()
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 2138, in run
    super().run()
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 844, in run
    while self.step():
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 754, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 2259, in RETURN_VALUE
    self.output.compile_subgraph(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 944, in compile_subgraph
    self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
  File "/home/pt-gpu/anaconda3/envs/zejun/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 1143, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/utils.py", line 265, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 1218, in call_user_compiler
    raise BackendCompilerFailed(self.compiler_fn, e).with_traceback(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 1199, in call_user_compiler
    compiled_fn = compiler_fn(gm, self.example_inputs())
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/repro/after_dynamo.py", line 117, in debug_wrapper
    compiled_gm = compiler_fn(gm, example_inputs)
  File "/home/pt-gpu/zejun/pytorch/torch/__init__.py", line 1739, in __call__
    return compile_fx(model_, inputs_, config_patches=self.config)
  File "/home/pt-gpu/anaconda3/envs/zejun/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/pt-gpu/zejun/pytorch/torch/_inductor/compile_fx.py", line 1360, in compile_fx
    return aot_autograd(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/backends/common.py", line 58, in compiler_fn
    cg = aot_module_simplified(gm, example_inputs, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/aot_autograd.py", line 903, in aot_module_simplified
    compiled_fn = create_aot_dispatcher_function(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/utils.py", line 265, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/aot_autograd.py", line 533, in create_aot_dispatcher_function
    fw_metadata = run_functionalized_fw_and_collect_metadata(
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/_aot_autograd/collect_metadata_analysis.py", line 555, in inner
    dynamic_dims = {
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/_aot_autograd/collect_metadata_analysis.py", line 556, in <setcomp>
    i for i, s in enumerate(o.shape) if not is_concrete_int(s)
  File "/home/pt-gpu/zejun/pytorch/torch/fx/experimental/symbolic_shapes.py", line 176, in is_concrete_int
    if isinstance(a.node.expr, sympy.core.numbers.Integer):
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
AttributeError: 'torch._C._SymNode' object has no attribute 'expr'

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True

```

### Versions

Collecting environment information...
PyTorch version: 2.4.0a0+gitb198423
Is debug build: False
CUDA used to build PyTorch: 12.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
Clang version: Could not collect
CMake version: version 3.27.2
Libc version: glibc-2.31

Python version: 3.9.17 (main, Jul  5 2023, 20:41:20)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-97-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 12.3.52
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA A100-PCIE-40GB
Nvidia driver version: 545.23.06
cuDNN version: Probably one of the following:
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.6.0
/usr/local/cuda-11.4/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.6.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      48 bits physical, 48 bits virtual
CPU(s):                             64
On-line CPU(s) list:                0-63
Thread(s) per core:                 1
Core(s) per socket:                 64
Socket(s):                          1
NUMA node(s):                       8
Vendor ID:                          AuthenticAMD
CPU family:                         25
Model:                              1
Model name:                         AMD EPYC 7713 64-Core Processor
Stepping:                           1
Frequency boost:                    enabled
CPU MHz:                            1500.000
CPU max MHz:                        3720.7029
CPU min MHz:                        1500.0000
BogoMIPS:                           4000.07
Virtualization:                     AMD-V
L1d cache:                          2 MiB
L1i cache:                          2 MiB
L2 cache:                           32 MiB
L3 cache:                           256 MiB
NUMA node0 CPU(s):                  0-7
NUMA node1 CPU(s):                  8-15
NUMA node2 CPU(s):                  16-23
NUMA node3 CPU(s):                  24-31
NUMA node4 CPU(s):                  32-39
NUMA node5 CPU(s):                  40-47
NUMA node6 CPU(s):                  48-55
NUMA node7 CPU(s):                  56-63
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Mitigation; safe RET
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm

Versions of relevant libraries:
[pip3] bert-pytorch==0.0.1a4
[pip3] clip-anytorch==2.6.0
[pip3] CoCa-pytorch==0.1.0
[pip3] dalle2-pytorch==1.14.2
[pip3] ema-pytorch==0.3.3
[pip3] flake8==3.8.2
[pip3] flake8-bugbear==20.1.4
[pip3] flake8-coding==1.3.3
[pip3] flake8-comprehensions==3.3.0
[pip3] flake8-executable==2.0.4
[pip3] flake8-logging-format==0.9.0
[pip3] flake8-pyi==20.5.0
[pip3] flake8-simplify==0.19.3
[pip3] functorch==1.14.0a0+b71aa0b
[pip3] mypy==1.4.1
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.23.5
[pip3] onnx==1.15.0
[pip3] open-clip-torch==2.24.0
[pip3] optree==0.9.1
[pip3] pytorch-triton==3.0.0+901819d2b6
[pip3] pytorch-warmup==0.1.1
[pip3] rotary-embedding-torch==0.3.3
[pip3] torch==2.4.0a0+gitb198423
[pip3] torch-fidelity==0.3.0
[pip3] torch_geometric==2.4.0
[pip3] torchaudio==2.2.0
[pip3] torchbench==0.1
[pip3] torchmetrics==1.3.0.post0
[pip3] torchmultimodal==0.1.0b0
[pip3] torchvision==0.16.0
[pip3] vector_quantize_pytorch==1.12.17
[conda] bert-pytorch              0.0.1a4                   dev_0    <develop>
[conda] blas                      1.0                         mkl  
[conda] clip-anytorch             2.6.0                    pypi_0    pypi
[conda] coca-pytorch              0.1.0                    pypi_0    pypi
[conda] dalle2-pytorch            1.14.2                   pypi_0    pypi
[conda] ema-pytorch               0.3.3                    pypi_0    pypi
[conda] functorch                 1.14.0a0+b71aa0b          pypi_0    pypi
[conda] mkl                       2023.1.0         h213fc3f_46343  
[conda] mkl-service               2.4.0            py39h5eee18b_1  
[conda] mkl_fft                   1.3.8            py39h5eee18b_0  
[conda] mkl_random                1.2.4            py39hdb19cb5_0  
[conda] numpy                     1.23.5                   pypi_0    pypi
[conda] numpy-base                1.26.0           py39hb5e798b_0  
[conda] open-clip-torch           2.24.0                   pypi_0    pypi
[conda] optree                    0.9.1                    pypi_0    pypi
[conda] pytorch-triton            3.0.0+901819d2b6          pypi_0    pypi
[conda] pytorch-warmup            0.1.1                    pypi_0    pypi
[conda] rotary-embedding-torch    0.3.3                    pypi_0    pypi
[conda] torch                     2.4.0a0+gitcf06189           dev_0    <develop>
[conda] torch-fidelity            0.3.0                    pypi_0    pypi
[conda] torch-geometric           2.4.0                    pypi_0    pypi
[conda] torchaudio                2.2.0                    pypi_0    pypi
[conda] torchbench                0.1                       dev_0    <develop>
[conda] torchfix                  0.4.0                    pypi_0    pypi
[conda] torchmetrics              1.3.0.post0              pypi_0    pypi
[conda] torchmultimodal           0.1.0b0                  pypi_0    pypi
[conda] torchvision               0.16.0                   pypi_0    pypi
[conda] vector-quantize-pytorch   1.12.17                  pypi_0    pypi

cc @cpuhrsch @jbschlosser @bhosmer @drisspg @soulitzer @ezyang

**Comment by jbschlosser** [comment]
Hey @zejun-chen, thanks for the report! Unfortunately, as you've pointed out, constructing a nested tensor within a torch.compiled region (i.e. as part of the graph) is unsupported at the moment for reasons related to symbolic shapes (see #118446). We are actively working on addressing this.

In the meantime, as a workaround, passing nested tensors as inputs to a torch.compiled function should work fine, but please let us know if you have troubles there too.

**Comment by zejun-chen** [comment]
Thank you @jbschlosser 
When passing nested tensor to torch.compile, the function is compiled successfully.

**Comment by cpuhrsch** [comment]
@zejun-chen - For our own education, if you want, is it ok if I ask you to add some context for your use of NestedTensor? This can help us prioritize and increase upcoming development.

**Comment by zejun-chen** [comment]
> @zejun-chen - For our own education, if you want, is it ok if I ask you to add some context for your use of NestedTensor? This can help us prioritize and increase upcoming development.

Thank you for attention.
I just write some simple cases to test NestedTensor compilation issues and found failure here for compilation of Jagged tensor creation. No real usage is in my workload but i will collect NestedTensor usage in my team.

**Comment by jbschlosser** [comment]
Closing as addressed in #130292.
